server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -207,5 +217,57 @@ public SimpleHealthIndicatorDetails getDetails() {
                 )
             );
         }
+
+        public List<HealthIndicatorImpact> getImpacts() {
+            final List<HealthIndicatorImpact> impacts = new ArrayList<>();
+            LinkedHashSet<String> indicesWithUnavailablePrimariesAndReplicas = new LinkedHashSet<>(primaries.indicesWithUnavailableShards);
+            indicesWithUnavailablePrimariesAndReplicas.retainAll(replicas.indicesWithUnavailableShards);
+            LinkedHashSet<String> indicesWithUnavailablePrimariesOnly = new LinkedHashSet<>(primaries.indicesWithUnavailableShards);
+            indicesWithUnavailablePrimariesOnly.removeAll(replicas.indicesWithUnavailableShards);
+            LinkedHashSet<String> indicesWithUnavailableReplicasOnly = new LinkedHashSet<>(replicas.indicesWithUnavailableShards);
+            indicesWithUnavailableReplicasOnly.removeAll(primaries.indicesWithUnavailableShards);
+            if (indicesWithUnavailablePrimariesAndReplicas.isEmpty() == false) {
+                String impactDescription = String.format(
+                    Locale.ROOT,
+                    "%s [%s] %s unavailable. You are at risk of losing the data in %s.",
```

Comment:
The current impacts are:

Primaries available, replicas unassigned:
```
"impact": {
  "description": "Redundancy for 3 indices [my-index-000001, my-index-000003, my-index-000002] is currently disrupted. Fault tolerance and search scalability are reduced.",
  "severity": 3
}
```

Primaries unassigned, but replicas available:
```
"impact": {
  "description": "Cannot add data to 1 index [my-index-000001]. Searches might return incomplete results.",
  "severity": 2
}
```

Both primaries and replicas unavailable:
```
 "impact": {
   "description": "Indices [my-index-000001, my-index-000003, my-index-000002] are unavailable. You are at risk of losing the data in these indices.",
   "severity": 1
}
```

@cristina-eleonora @shubhaat @Leaf-Lin Can you please provide some feedback on these messages?

Reply:
Thanks for the ping @andreidan !


> Primaries available, replicas unassigned:
> 
> ```
> "impact": {
>   "description": "Redundancy for 3 indices [my-index-000001, my-index-000003, my-index-000002] is currently disrupted. Fault tolerance and search scalability are reduced.",
>   "severity": 3
> }
> ```
 
Could we rephrase the first half to something like: `Data redundancy is affected on 3 indices  [my-index-000001, my-index-000003, my-index-000002]. `? Are there any impacts on search/ingest? And could we phrase `Fault tolerance and search scalability are reduced.` in a way that a non technical / minimally technical person would understand? Perhaps some of the writers could help with this.

> Primaries unassigned, but replicas available:
> 
> ```
> "impact": {
>   "description": "Cannot add data to 1 index [my-index-000001]. Searches might return incomplete results.",
>   "severity": 2
> }
> ```

Looks great 👍🏻 


> Both primaries and replicas unavailable:
> 
> ```
>  "impact": {
>    "description": "Indices [my-index-000001, my-index-000003, my-index-000002] are unavailable. You are at risk of losing the data in these indices.",
>    "severity": 1
> }
> ```
> 

Does this mean that those indices cannot be accessed anymore? Or that the data on those indices cannot be accessed? How will that impact search/ingest?

Reply:
I thought `Primaries unassigned, but replicas available` would not happen because replica will be promoted?

Reply:
Based on @Leaf-Lin comment above we decided to simplify the impacts into the following categories:
1. primaries unavailable for indices:
```
Cannot add data to 1 index [my-index-000001]. Searches will return incomplete results.
```
2. replicas unavailable for indices:
```
Data redundancy is affected on 3 indices [my-index-000001, my-index-000003, my-index-000002]. Fault tolerance and search scalability are reduced.
```

Thanks for your thoughts on the messages @cristina-eleonora 
I think with the simplified cases we have the `Fault tolerance and search scalability are reduced.` bit to rephrase somehow. 
When talking about replicas in our [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/scalability.html) we say:
```
Replicas provide redundant copies of your data to protect against hardware failure and increase capacity to serve read requests like searching or retrieving a document
``` 

Would impact number 2 make more sense if it was: (or something along those lines?)
```
Data redundancy is affected on 3 indices [my-index-000001, my-index-000003, my-index-000002]. The capacity to serve search requests is reduced and you could be susceptible to data loss in case of hardware failure.
```

Reply:
OK I have the wording in from Andre's last comment. I can easily change it if we need to.

Reply:
Thanks @andreidan. I have a few comments: 

1. For every impact, I think it will be more readable with the second part being in the front such that `You are at risk of losing the data in the indices. Indices [my-index-000001, my-index-000003, my-index-000002] are unavailable.` This makes the impact statement _before_ the cause. 
2. I would love to make the one for `Primaries available, replicas unassigned` a bit more friendly, but I am at a loss, it does make sense the way it is in terms of fault tolerance and search scalability, but maybe it can be `Fault tolerance impacted, and search will not scale ....`

Reply:
Thanks for revisiting these, @andreidan !

The term I struggle with the most is `search scalability` -  I have a hard time figuring out (myself) what that means, and to Shubha's point, I wonder if we can phrase it in a way that's easier to understand. Perhaps @elastic/platform-writers can help?

++ on @shubhaat 's suggestion on leading with the impact and listing the cause after.

Reply:
For missing replicas, how about just:
`Searches might return slower than expected. Fewer redundant copies of the data exist than expected on 3 indices [my-index-000001, my-index-000003, my-index-000002].`
Or maybe the slightly scarier:
`Searches might return slower than expected, and you are at higher risk of the cluster becoming unstable in the future. Fewer redundant copies of the data exist than expected on 3 indices [my-index-000001, my-index-000003, my-index-000002].`
Or the even scarier:
`Searches might return slower than expected, and you are at higher risk of losing data. Fewer redundant copies of the data exist than expected on 3 indices [my-index-000001, my-index-000003, my-index-000002].`

Reply:
Pinging @alaudazzi , our UX writer.

Reply:
I went with Andre's last wording, but can update at any point later if we work out something better.